apps/browser/qml/pages/components/PopUpMenu.qml

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2021 Open Mobile Platform LLC.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import QtQuick 2.2
+import QtGraphicalEffects 1.0
+import Sailfish.Silica 1.0
+
+Item {
+    id: popUpMenu
+
+    property var menuItem
+    property var footer
+    property bool active
+    readonly property int cornerRadius: 12
+    readonly property int widthRatio: 18
+    readonly property int heightRatio: 28
+    readonly property string _tablet: "tablet"
+    readonly property string _phone: "phone"
+    readonly property bool isTablet: Screen.sizeCategory > Screen.Medium
+    readonly property bool _open: active
+
+    signal closed
+
+    width: Theme.paddingLarge * widthRatio
```

Comment:
Running on an Xperia XA2, the width of the menu in lanscape mode is too thin to properly fit the four footer icons. It looks like there needs to be a minimum width here of `(4 * overlay.toolBar.iconWidth) + (3 * (Theme.paddingLarge + Theme.paddingSmall + Theme.paddingSmall / 2)) + (2 * Theme.paddingMedium)` (assuming four icons, three gaps and a margin either side), or something close to that.

Reply:
Thanks, changed the logic for the width. Unfortunately, I cannot check on a device with pixelRatio> 1.25